Tests/SwiftAlgorithmsTests/RotateTests.swift

```diff
@@ -10,9 +10,43 @@
 //===----------------------------------------------------------------------===//
 
 import XCTest
-import Algorithms
+@testable import Algorithms
 
 final class RotateTests: XCTestCase {
+  func testUnderscoreReverse() {
+    var input = ["a", "b", "c", "d", "e", "f", "g", "h", "i", "j", "k", "l", "m", "n", "o", "p"]
+    let limit: Int = 4
+    let (lower, upper) = input._reverse(subrange: input.startIndex..<input.endIndex, until: input.startIndex.advanced(by: limit))
+    let expected = ["p", "o", "n", "m", "e", "f", "g", "h", "i", "j", "k", "l", "d", "c", "b", "a"]
+    XCTAssertEqual(input, expected)
+    XCTAssertEqual(lower, input.startIndex.advanced(by: limit))
+    XCTAssertEqual(upper, input.endIndex.advanced(by: -limit))
+    
+    var numbers = [10, 20, 30, 40, 50, 60, 70, 80]
+    numbers.reverse(subrange: 0..<4)
+    XCTAssertEqual(numbers, [40, 30, 20, 10, 50, 60, 70, 80])
+  }
+  
+  func testReverse() {
+    var numbers = [10, 20, 30, 40, 50, 60, 70, 80]
+    numbers.reverse(subrange: 0..<4)
+    XCTAssertEqual(numbers, [40, 30, 20, 10, 50, 60, 70, 80])
+  }
+  
+  func testRotateSubrangeExample() {
+    var numbers = [10, 20, 30, 40, 50, 60, 70, 80]
+    let oldStart = numbers.rotate(subrange: 0..<4, toStartAt: 2)
+    XCTAssertEqual(numbers, [30, 40, 10, 20, 50, 60, 70, 80])
+    XCTAssertEqual(numbers[oldStart], 10)
+  }
+  
+  func testRotateExample() {
+    var numbers = [10, 20, 30, 40, 50, 60, 70, 80]
+    let oldStart = numbers.rotate(toStartAt: 3)
+    XCTAssertEqual(numbers, [40, 50, 60, 70, 80, 10, 20, 30])
+    XCTAssertEqual(numbers[oldStart], 10)
+  }
```

Comment:
Instead of these, do you want to take a crack at comprehensively rotating the subranges of a short array?